### PR TITLE
SNOW-1728002 get rid of reflection when enabling iceberg streaming

### PR DESCRIPTION
**Pull request by sfc-gh-bzabek**
# Overview

related to SNOW-1728002
It's temporary approach. Finally we will set iceberg mode by invoking SnowflakeStreamingIngestClientFactory.builder.setIceberg(true) but the method is not yet public.

## Pre-review checklist
- [ ] This change should be part of a Behavior Change Release. See [go/behavior-change](http://go/behavior-change).
- [ ] This change has passed Merge gate tests
- [ ] Snowpipe Changes
- [ ] Snowpipe Streaming Changes
- [ ] This change is TEST-ONLY
- [ ] This change is README/Javadocs only
- [ ] This change is protected by a config parameter <PARAMETER_NAME> eg `snowflake.ingestion.method`.
    - [ ] `Yes` - Added end to end and Unit Tests. 
    - [ ] `No` - Suggest why it is not param protected
- [ ] Is his change protected by parameter <PARAMETER_NAME> on the server side?
    - [ ] The parameter/feature is not yet active in production (partial rollout or PrPr, see [Changes for Unreleased Features and Fixes](http://go/ppp-prpr)).
    - [ ] If there is an issue, it can be safely mitigated by turning the parameter off. This is also verified by a test (See [go/ppp](http://go/ppp)).

### CODE REVIEW

File: src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProperties.java [hunk 1]
```diff
@@ -90,6 +90,11 @@ public StreamingClientProperties(Map<String, String> connectorConfig) {
 
     // Override only if the streaming client properties are explicitly set in config
     this.parameterOverrides = new HashMap<>();
+    if (isIcebergEnabled) {
+      // TODO should be replaced by SnowflakeStreamingIngestClientFactory.builder.setIceberg(true)
+      // call in SNOW-1728002
+      this.parameterOverrides.put("enable_iceberg_streaming", "true");
+    }
```
Comment (sfc-gh-bzabek) [inline src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProperties.java:97]: By the way the param is set in IcebergIngestionIT setUp.

**Comment by sfc-gh-bzabek** [comment]
I see my commits name are bad, i'll change them while squashing.

File: src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProperties.java (hunk 1 above)
Comment (sfc-gh-bzabek) [reply src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProperties.java:97]: There are no enum for "enable_iceberg_streaming" in ingest-sdk.

Comment (sfc-gh-mbobowski) [reply src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProperties.java:97]: I found one. Please try: `ParameterProvider.ENABLE_ICEBERG_STREAMING`

Comment (sfc-gh-mbobowski) [reply src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProperties.java:97]: Additionally please remove the TODO comment and add `SNOW-1728002` to the PR name. This is everything that needs to be done in this scope.

Comment (sfc-gh-bzabek) [reply src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProperties.java]: > I found one. Please try: ParameterProvider.ENABLE_ICEBERG_STREAMING

I had to update ingest-sdk for it.

> Additionally please remove the TODO comment and add SNOW-1728002 to the PR name. This is everything that needs to be done in scope of this JIRA.

Done

Comment (sfc-gh-bzabek) [reply src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProperties.java:95]: When I used variable from ParameterProvider it fails to compile, I think ingest-sdk 3.0.0 is not yet available

**Review by sfc-gh-mbobowski — APPROVED** [review]

**Pull request merged**